### PR TITLE
MTP-1976: Add api to manage monitored partial email addresses

### PR DESCRIPTION
**Pull request by ushkarev**

### CODE REVIEW

File: mtp_api/apps/security/models.py [hunk 1]
```diff
@@ -286,6 +288,23 @@ def __str__(self):
...
+    class Meta:
+        ordering = ('keyword',)
+        verbose_name = 'monitored partial email address'
+        verbose_name_plural = 'monitored partial email addresses'
+
+    def __str__(self):
+        return self.keyword
+
+    def matches(self, email_address: str) -> bool:
+        return self.keyword in email_address.lower()
```
Comment (xoen) [inline mtp_api/apps/security/models.py:305]: should we lowecase `self.keyword` as well? Unless you're planning to store these in lowecase?

Comment (ushkarev) [reply mtp_api/apps/security/models.py:305]: it happens here: https://github.com/ministryofjustice/money-to-prisoners-api/pull/769/commits/f894533b3b96f234c6c6efd87405cd52d07b61fc#diff-3cf42ec26dec1e69b6d44a59bb6095460400d603136ea39c825fd17666fa2a86R501

Comment (ushkarev) [reply mtp_api/apps/security/models.py:305]: admittedly, any created through django admin could be incorrect (but nobody should mess around in there anyway)

File: mtp_api/apps/mtp_auth/fixtures/initial_groups.json [hunk 2]
```diff
@@ -131,7 +131,10 @@
       ["change_checkautoacceptrule", "security", "checkautoacceptrule"],
       ["add_checkautoacceptrulestate", "security", "checkautoacceptrulestate"],
       ["view_checkautoacceptrulestate", "security", "checkautoacceptrulestate"],
-      ["change_checkautoacceptrulestate", "security", "checkautoacceptrulestate"]
+      ["change_checkautoacceptrulestate", "security", "checkautoacceptrulestate"],
+      ["add_monitoredpartialemailaddress", "security", "monitoredpartialemailaddress"],
+      ["delete_monitoredpartialemailaddress", "security", "monitoredpartialemailaddress"],
+      ["view_monitoredpartialemailaddress", "security", "monitoredpartialemailaddress"]
```
Comment (xoen) [inline mtp_api/apps/mtp_auth/fixtures/initial_groups.json:137]: could be good to add the trailing comma there :)

**Review by xoen — APPROVED** [review]

File: mtp_api/apps/mtp_auth/fixtures/initial_groups.json (hunk 2 above)
Comment (ushkarev) [reply mtp_api/apps/mtp_auth/fixtures/initial_groups.json:137]: i’d LOVE to but it’s json :(

Comment (xoen) [reply mtp_api/apps/mtp_auth/fixtures/initial_groups.json:137]: ahaha darn it, every single time. Ignore 😝

**Pull request merged**